templates/spec/system/search_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'solidus_starter_frontend_helper'
+
+RSpec.describe 'searching products', type: :system do
+  include_context "custom products"
+
+  before(:each) do
+    visit root_path
+  end
+
+  it 'falls back to normal search for plain HTML requests' do
+    fill_in 'keywords', with: 'shirt'
+    click_button 'Search'
+
+    expect(page.all('ul.products-grid li').size).to eq(1)
+  end
+
+  context 'with JS enabled', :js do
+    it 'shows autocomplete suggestions' do
+      fill_in 'keywords', with: 'ruby'
+      expect(page.all('ul.autocomplete-results__list li.autocomplete-results__item:not(.autocomplete-results__item--empty)').size).to eq(8)
```

Comment:
I'm a bit concerned about how these tests rely so heavily on classes used by CSS. Thoughts about leveraging `data-` attributes?

Reply:
Always unsure about how to handle this problem, but I like the idea, especially if we are testing some JS-driven behavior that will change when we change those `data-` attributes. Do you know if there's any best practice for what to use in specs?

Reply:
> Do you know if there's any best practice for what to use in specs?

Not sure. If possible, we should rely on behavior instead of concrete UI implementation. If not possible, I think we should try to leverage HTML semantics and, if that's neither possible, I sometimes have used `data-test` attribute names. I think there's no perfect solution, though 😢

Reply:
Done. I just left a check on the "current" CSS class on results, which is what the JS does to mark the currently selected item.